sdk/src/beta9/abstractions/base/runner.py

```diff
@@ -197,15 +199,23 @@ def _map_callable_to_attr(self, *, attr: str, func: Callable):
             return
 
         module = inspect.getmodule(func)  # Determine module / function name
-        if module:
+        if hasattr(module, "__file__"):
             module_file = os.path.relpath(module.__file__, start=os.getcwd()).replace("/", ".")
             module_name = os.path.splitext(module_file)[0]
+        elif in_jupyter():
+            tmp_file = create_tmp_jupyter_file(module._ih)
+            module_name = tmp_file.name.split("/")[-1].removesuffix(".py")
+            self.tmp_files.append(tmp_file)
         else:
             module_name = "__main__"
 
         function_name = func.__name__
         setattr(self, attr, f"{module_name}:{function_name}")
 
+    def remove_tmp_files(self) -> None:
```

Comment:
since this isn't called outside of the class, I would prefix it with _ (e.g. `_remove_tmp_files()`)